businessRules/users.js

```diff
@@ -18,70 +25,38 @@ function registerUser(userName, password, firstName, lastName) {
 }
 
 function login(userName, password) {
-    if (!userName) {
-        throw new Error('Error. There was no username provided.');
-    }
-    if (!password) {
-        throw new Error('Error. There was no password provided.');
-    }
-
-    let users = getUsers();
+    validateUserName(userName);
 
-    const user = users.find(
-        (user) => user.userName === userName && user.password === password
-    );
+    validatePassword(password);
 
-    if (!user) {
-        throw new Error('Error. No such user.');
-    }
+    const user = findUserWithSameUserNameAndPassword(userName, password);
 
     const sessionId = md5(v4()); //Should be REAL cryptography.
+
     updateUser(user.id, 'session', sessionId);
```

Comment:
The readability of this function call can be improved. Instead of passing `session` string, you can pass: `setSessionId` - if that's what this function call does

Reply:
I think I've understood, tryed to improve it.